src/loki/polygon_search.cc

```diff
@@ -0,0 +1,158 @@
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/geometries/register/ring.hpp>
+
+#include <valhalla/loki/polygon_search.h>
+#include <valhalla/midgard/constants.h>
+#include <valhalla/midgard/pointll.h>
+#include <valhalla/worker.h>
+
+namespace bg = boost::geometry;
+namespace vm = valhalla::midgard;
+namespace vb = valhalla::baldr;
+namespace vl = valhalla::loki;
+
+BOOST_GEOMETRY_REGISTER_POINT_2D(valhalla::midgard::PointLL,
+                                 double,
+                                 bg::cs::geographic<bg::degree>,
+                                 first,
+                                 second)
+BOOST_GEOMETRY_REGISTER_RING(std::vector<vm::PointLL>)
+
+namespace {
+// register a few boost.geometry types
+using line_bg_t = bg::model::linestring<vm::PointLL>;
+using ring_bg_t = std::vector<vm::PointLL>;
+
+// map of tile for map of bin ids & their ring ids
+using bins_collector =
+    std::unordered_map<uint32_t, std::unordered_map<unsigned short, std::vector<size_t>>>;
```

Comment:
i believe we could simplify this a bit but we can just leave that as a TODO for future work. fine the way it is for now